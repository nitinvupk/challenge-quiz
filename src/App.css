body {
  margin: 0;
  padding: 0;
}
.container {
  width: 40%;  
  min-height: 500px;
  min-width: 500px;
  margin: 60px auto;
  border: 5px solid #ececec;
}
.questionBox {
  padding: 50px;
}
.ratingDisable {
  color: #cbcccb;
}
.buttonBox {
  display: flex;
  justify-content: space-between;
  margin-bottom: 35px;
}
.btn {
  background: #e5e6e5;
  border: 1px solid #000;
  border-radius: 4px;
  padding: 6px 0;
  text-align: center;
  color: #000;
  cursor: pointer;
  font-size: 16px;
  width: 200px;
}
.selected {
  color: #fff;
  background: #000;
}
.disable {
  cursor: not-allowed;
  opacity: 0.4;
}
.text-center {
  text-align: center;
}