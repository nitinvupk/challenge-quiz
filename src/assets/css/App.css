body {
  margin: 0;
  padding: 0;
}
.container {
  width: 40%;  
  min-height: 500px;
  min-width: 500px;
  margin: 60px auto;
  border: 5px solid #ececec;
}
.questionBox {
  padding: 15px 50px;
}
.ratingDisable {
  color: #cbcccb;
}
.buttonBox {
  display: flex;
  justify-content: space-between;
  margin-bottom: 35px;
  flex-wrap: wrap;
}
.btnContainer {
  width: 50%;
  margin-bottom: 15px;
}
.btn {
  background: #e5e6e5;
  border: 1px solid #000;
  border-radius: 4px;
  padding: 6px 0;
  text-align: center;
  color: #000;
  cursor: pointer;
  font-size: 16px;
  width: 200px;
}
.selected {
  color: #fff;
  background: #000;
}
.disable {
  cursor: not-allowed;
  opacity: 0.4;
}
.text-center {
  text-align: center;
}
.scoreBox {
  border: 1px solid #000;
  border-radius: 4px;
  height: 25px;
  display: flex;
  width: 100%;
}
.lowestPossibleBar {
  background: #000;
  float: left;
  height: 100%;
}
.maximumPercentBar {
  background: #d2d2d2;
  float: left;
  height: 100%;
}
.currentScoreBar {
  background: #717171;
  float: left;
  height: 100%;
}
.percentBar {
  height: 25px;
}
.currentScorePercent {
  color: #494949;
  font-size: 18px;
  padding: 8px;
  float: left;
}
.maxScorePercent {
  color: #494949;
  font-size: 18px;
  padding: 8px;
  float: right;
}
.loadingBar {
  width: 100%;
  height: 15px;
  background: #fff;
}
.progress {
  display: block;
  background: #a9aaa9;
  height: 100%
}