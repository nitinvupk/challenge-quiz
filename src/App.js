import React, { Fragment, useState, useEffect } from 'react'
import { calculatePercent, shuffle } from './helpers/index'
import allQuestions from './assets/questions.json'
import { RatingAndCategory, Question, ScoreBar, QuestionLoadingBar } from './components'
import './assets/css/App.css'

function App () {
  const [ questions, setQuestions ] = useState([])
  const [ selectedQuestion, setSelectedQuestion ] = useState({})
  const [ currentQueNumber, setCurrentQueNumber ] = useState(0)
  const [ finished, setFinished ] = useState(false)
  const [ score, setScore ] = useState(0)

  useEffect(() => {
    setQuestions(allQuestions)
  }, [])

  useEffect(() => {
    if (questions.length) {
      handleSelectQuestion()
    }
  }, [questions, currentQueNumber])

  const handleSelectQuestion = () => {
    let selectedQuestion = questions[currentQueNumber]
    let arrAns = []
    arrAns = selectedQuestion.incorrect_answers
    arrAns.push(selectedQuestion.correct_answer)
    arrAns = shuffle(arrAns)
    selectedQuestion['options'] = arrAns
    setSelectedQuestion(selectedQuestion)
  }

  const handleNextQuestion = () => {
    let nextNumber = currentQueNumber + 1
    if (nextNumber < questions.length) {
      setCurrentQueNumber(nextNumber)
    } else {
      setFinished(true)
    }
  }

  const handleScore = () => {
    setScore(score + 1)
  }

  let { category, difficulty, question, correct_answer, options } = selectedQuestion

  let ratingProps = {
    currentQueNumber,
    totalCount: questions.length,
    category,
    difficulty
  }

  let questionProps = {
    question,
    correct_answer,
    options,
    nextQuestion: handleNextQuestion,
    handleScore
  }

  let scoreProps = {
    score,
    currentQueNumber,
    totalCount: questions.length
  }

  let loadingBarProps = {
    questionNumber: currentQueNumber + 1,
    totalCount: questions.length
  }

  return (
    <div className='container'>
      <QuestionLoadingBar {...loadingBarProps} />
      <div className='questionBox'>
        {!finished &&
          <Fragment>
            <RatingAndCategory {...ratingProps} />
            <Question {...questionProps} />
            <ScoreBar {...scoreProps} />
          </Fragment>
        }
        {finished &&
          <Fragment>
            <h1>Your Score: { calculatePercent(score, questions.length) }%</h1>
            <h1>Test Completed!</h1>
          </Fragment>
        }
      </div>
    </div>
  )
}

export default App
