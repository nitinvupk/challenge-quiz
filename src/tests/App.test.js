import React from 'react'
import { shallow } from 'enzyme'
import App from '../App'

describe('App', () => {
  it('should render app component', () => {
    const wrapper = shallow(<App debug />)
    expect(wrapper).toMatchSnapshot()
  })
})
