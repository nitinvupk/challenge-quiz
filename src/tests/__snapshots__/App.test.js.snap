// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render app component 1`] = `
<div
  className="container"
>
  <QuestionLoadingBar
    questionNumber={1}
    totalCount={0}
  />
  <div
    className="questionBox"
  >
    <RatingAndCategory
      currentQueNumber={0}
      totalCount={0}
    />
    <Question
      handleScore={[Function]}
      nextQuestion={[Function]}
    />
    <ScoreBar
      currentQueNumber={0}
      score={0}
      totalCount={0}
    />
  </div>
</div>
`;
